roxctl/declarativeconfig/create/auth_provider.go

```diff
@@ -0,0 +1,294 @@
+The expected format is <endpoint>:<port>`)
+
+	cmd.PersistentFlags().StringToStringVar(&authProviderCmd.requiredAttributes, "required-attributes",
+		map[string]string{}, `list of attributes that are required to be returned by the auth provider during authentication,
+e.g. --required-attributes "my_org=sample-org"`)
+
+	// pflag.FlagSet currently lacks to provide a tuple of repeated key value pairs, hence we need to resort to
+	// providing three separate flags which hold the values required to construct a single group:
+	// --groups-key, --groups-value, --group-role.
+	// They can be repeated as such within the CLI flags:
+	// --groups-key "email" --groups-value "my@domain.com" --groups-value"Admin" \
+	// --groups-key "user" --groups-value "id" --groups-role "Analyst"
+	cmd.PersistentFlags().StringSliceVar(&authProviderCmd.groupsKeys, "groups-key", []string{},
+		`keys of the groups to add within the auth provider. Note that the tuple of key, value, role should
+be of the same length.
+Example of a group: --groups-key "email" --groups-value "my@domain.com" --groups-role "Admin"`)
+
+	cmd.PersistentFlags().StringSliceVar(&authProviderCmd.groupsValues, "groups-value", []string{},
+		`values of the groups to add within the auth provider. Note that the tuple of key, value, role should
+be of the same length.
+Example of a group: --groups-key "email" --groups-value "my@domain.com" --groups-role "Admin"`)
+	cmd.PersistentFlags().StringSliceVar(&authProviderCmd.groupsRoles, "groups-role", []string{},
+		`role of the groups to add within the auth provider. Note that the tuple of key, value, role should
+be of the same length.
+Example of a group: --groups-key "email" --groups-value "my@domain.com" --groups-role "Admin"`)
+
+	cmd.MarkFlagsRequiredTogether("name", "ui-endpoint")
+
+	cmd.AddCommand(
+		authProviderCmd.oidcCommand(),
+		authProviderCmd.samlCommand(),
+		authProviderCmd.iapCommand(),
+		authProviderCmd.userPKICommand(),
+		authProviderCmd.openShiftCommand(),
+	)
+
+	return cmd
+}
+
+type authProviderCmd struct {
+	authProvider *declarativeconfig.AuthProvider
+
+	requiredAttributes map[string]string
+	claimMapping       map[string]string
+
+	groupsKeys   []string
+	groupsValues []string
+	groupsRoles  []string
+
+	oidcConfig    *declarativeconfig.OIDCConfig
+	samlConfig    *declarativeconfig.SAMLConfig
+	iapConfig     *declarativeconfig.IAPConfig
+	userPKIConfig *declarativeconfig.UserpkiConfig
+
+	// Custom mappings for SAML command.
+	samlIDPCertFile string
+
+	// Custom mapping for User PKI command.
+	userPKICAFile string
+
+	env environment.Environment
+}
+
+func (a *authProviderCmd) oidcCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "oidc",
+		Args:  cobra.NoArgs,
+		RunE:  a.RunE(),
+		Short: "Create a declarative configuration for an OIDC auth provider",
+	}
+	a.oidcConfig = &declarativeconfig.OIDCConfig{}
+
+	cmd.Flags().StringVar(&a.oidcConfig.Issuer, "issuer", "",
```

Comment:
Can we require `Issuer`, `CallbackMode` and `ClientID`? I think we always need them to create proper OIDC auth provider

Reply:
I think we can leave `CallbackMode` out since it will default to `auto`.